examples/osggpucull/osggpucull.cpp

```diff
@@ -858,33 +853,31 @@ struct ResetTexturesCallback : public osg::StateSet::Callback
         texUnitsDirty.push_back(texUnit);
     }
 
-    void addTextureDirtyParams( unsigned int texUnit )
-    {
-        texUnitsDirtyParams.push_back(texUnit);
-    }
-
     virtual void operator() (osg::StateSet* stateset, osg::NodeVisitor* /*nv*/)
     {
         std::vector<unsigned int>::iterator it,eit;
         for(it=texUnitsDirty.begin(), eit=texUnitsDirty.end(); it!=eit; ++it)
-        {
-            osg::TextureBuffer* tex = dynamic_cast<osg::TextureBuffer*>( stateset->getTextureAttribute(*it,osg::StateAttribute::TEXTURE) );
-            if(tex==NULL)
-                continue;
-            osg::BufferData* img =const_cast<osg::BufferData*>(tex->getBufferData());
-            if(img!=NULL)
-                img->dirty();
-        }
-        for(it=texUnitsDirtyParams.begin(), eit=texUnitsDirtyParams.end(); it!=eit; ++it)
-        {
-            osg::TextureBuffer* tex = dynamic_cast<osg::TextureBuffer*>( stateset->getTextureAttribute(*it,osg::StateAttribute::TEXTURE) );
-            if(tex!=NULL)
-                tex->dirtyTextureParameters();
+    {
+           osg::BindImageTexture*imb = 0;
+           const osg::StateSet::AttributeList & texlist= stateset->getTextureAttributeList()[*it];
```

Comment:
this dirty part is because ss.getTexAttribute(tu,type) fails : member searched is always set to 0 !?